### PR TITLE
Update doc layout

### PR DESCRIPTION
**Pull request by bjrara**
Notes:
1. Deleted `docs/example/prometheus.yaml`. This file was committed on Sep 23, 2020, and there was no updates since the. Reference check also showed no doc or resources was using it.
2. Migrated `eks-containerinsight-javajmx-sample` to https://github.com/aws-observability/aws-otel-test-framework/pull/283.
3. This PR doesn't include renaming `/deployment-template/eks/otel-container-insight.yaml` which will be done in https://github.com/aws-observability/aws-otel-collector/pull/458

### CODE REVIEW

**Review by pxaws — COMMENTED** [review]
I feel like the name `eks-sidecar.yaml` makes more sense to me than `otel-cloudwatch-deployment.yaml`. `otel-cloudwatch-deployment.yaml` doesn't make it clear about what this yaml does. 

Could you rename the file `otel-container-insights-deployment.yaml` to `otel-container-insight-prometheus.yaml` and the new file `eks-ci-daemonset.yaml` to `otel-container-insight.yaml`?

**Comment by mxiamxia** [comment]
> I feel like the name `eks-sidecar.yaml` makes more sense to me than `otel-cloudwatch-deployment.yaml`. `otel-cloudwatch-deployment.yaml` doesn't make it clear about what this yaml does.
> 
> Could you rename the file `otel-container-insights-deployment.yaml` to `otel-container-insight-prometheus.yaml` and the new file `eks-ci-daemonset.yaml` to `otel-container-insight.yaml`? (I did the renaming on another PR: #455 and we can discuss to come up with better names)

+1. should we do `otel-container-insight-prometheus.yaml` for containerinsight prometheus and `otel-container-insight-infra.yaml` for regular containerinsight metrics?

File: docs/developers/eks-demo.md [hunk 1]
```diff
@@ -55,11 +55,11 @@ export CLUSTER_NAME=<eks-cluster-name>
 export AWS_REGION=<aws-region>
 ```
 
-3. Deploy the application. An example config template can be found [here](../../deployment-template/eks/eks-sidecar.yaml).
+3. Deploy the application. An example config template can be found [here](../../deployment-template/eks/otel-cloudwatch-deployment.yaml).
```
Comment (mxiamxia) [inline docs/developers/eks-demo.md:58]: should we keep `sidecar` keyword? eg `otel-cloudwatch-sidecar.yaml`

**Comment by bjrara** [comment]
> Could you rename the file `otel-container-insights-deployment.yaml` to `otel-container-insight-prometheus.yaml`

Just to make sure I understand it correctly, is `insights` a typo on the official website? https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights-Prometheus-Sample-Workloads-appmesh.html

File: docs/developers/eks-demo.md (hunk 1 above)
Comment (bjrara) [reply docs/developers/eks-demo.md]: Done

**Comment by bjrara** [comment]
@mxiamxia @pxaws Thanks for the suggestions! I think I addressed all the comments, please help review!

**Comment by pxaws** [comment]
> > Could you rename the file `otel-container-insights-deployment.yaml` to `otel-container-insight-prometheus.yaml`
> 
> Just to make sure I understand it correctly, is `insights` a typo on the official website? For example: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights-Prometheus-Sample-Workloads-appmesh.html

I think `Container Insights` is the official name. But I guess it is still fine to use container insight

**Review by pxaws — APPROVED** [review]

**Pull request merged**